shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py

```diff
@@ -57,7 +57,15 @@ def on_submit(self):
 		frappe.db.set(self, 'fedex_status', ShipmentNoteOperationalStatus.InProgress)
 		frappe.db.set(self, 'tracking_number', self.box_list[0].tracking_number)
 
-	# def on_cancel(self):
+	def on_cancel(self):
+		tracking_ids_list = frappe.get_value("Sales Order", self.sales_order, "tracking_ids").split(',')
+
+		for box in self.box_list:
+			if box.tracking_number in tracking_ids_list:
+				tracking_ids_list.remove(box.tracking_number)
```

Comment:
We need to remove the tracking_number not present in the box_list right ??

Reply:
So when we cancel the shipment, the corresponding Tracking number(s) should be removed from the Sales Order Tracking IDs Field